nemo_skills/training/nemo_rl/start_sft.py

```diff
@@ -152,6 +153,19 @@ def sft_preprocessor(
         add_generation_prompt=add_generation_prompt,
     )
 
+    # ==================== START: BLOCK FOR DEBUGGING ====================
+    global _call_counter
+    if _call_counter < 1:  # Only print for the first 3 samples
+        print(f"\n--- 🐛 Debugging invocation #{_call_counter + 1} (for original sample idx: {idx}) ---")
+        # Loop through up to the first 3 messages in the log
+        for i, message in enumerate(message_log[:3]):
+            print(f"  Message [{i}]:")
+            print(f"    Role    : {message['role']}")
+            print(f"    Content : {message['content']}")
```

Comment:
Could you please add print for token ids as well? Those are also very important to inspect

Reply:
@Kipok need your suggestion. We load tokenizer as:
```
tokenizer = get_tokenizer(config["policy"]["tokenizer"])
```

So, we need to set `config["policy"]["tokenizer"]["chat_template"]="default"` if we see the data is in messages format. But we load the data later in `setup_data` method. So, how should we check if the data in messages format.

Given people may have data in different formats:
- input/output with template applied
- raw messages format
- messages format with template applied

I think it is best to just leave on the user to decide the template during submitting job as:
```
f'++policy.tokenizer.chat_template="default" '
```
?

Reply:
@coderabbitai, do you have suggestions for @wasiahmad on this? Assuming we still want to change the default to work as expected for both input and messages format (and ignore the case when messages are with template applied, assuming there people will provide appropriate parameters as this is not commonly used)

Reply:
that's not going to work. Let me push a change for this and @wasiahmad you can test on your side